### ISSUE TITLE
nerdctl compose sometimes hides error message (only print `exit status 1`)

### ISSUE THREAD

**Issue by djdongjin** [containerd/nerdctl#1879]
### Description

Often `nerdctl compose COMMAND` uses underlying `nerdctl COMMAND` to finish the job for each services (e.g., `nerdctl compose up` uses `nerdctl run` for each service container to run/start them). This is done by constructing and running `exec.Cmd` (as the example in `compose up/run` below).

https://github.com/containerd/nerdctl/blob/ded43fc8bd3fa278bc72c289cbbdabee46e656e0/pkg/composer/up_service.go#L154-L173

This causes that, when underlying `nerdctl COMMAND` has errors, `exec.Cmd` only returns `exit status 1` to `compose` instead of the real error message.

### Steps to reproduce the issue

> A side note: not sure why this is an error in nerdctl/containerd if the folder doesn't exist. docker seems just creates the folder if not exist.

```shell
# the service binds a volume to a non-exist folder
$ cat docker-compose.yaml
services:
  db:
    image: alpine
    volumes:
      - ./nonexist/folder:/var/lib/postgresql/db:rw
# compose only prints `exit status 1`
$ sudo nerdctl compose up
...
FATA[0000] error while creating container comproot_db_1: exit status 1
# run prints the real error
$ snd run -v=./nonexist/folder:/var/lib/postgresql/db alpine
...
FATA[0000] failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/home/ec2-user/code/tmp/comproot/nonexist/folder" to rootfs at "/var/lib/postgresql/db": stat /home/ec2-user/code/tmp/comproot/nonexist/folder: no such file or directory: unknown
```

### Describe the results you received and expected

For compose command we should prints the error message returned from underlying `nerdctl COMMAND`.

One solution might be to bind cmd.stderr to a buffer and if there is error returned, and we check/print the error in the buffer, for example:

```golang
cmd := exec.Command("sleep", "fafd")
var out bytes.Buffer
var stderr bytes.Buffer
cmd.Stdout = &out
cmd.Stderr = &stderr
err := cmd.Run()
fmt.Printf("cmd error: %v\nstdout: %s\nstderr: %s\n", err, out.String(), stderr.String())

// Output:
cmd error: exit status 1
stdout: 
stderr: sleep: invalid number 'fafd'
```

> This may be complicated in cases (run/exec/etc) where we need to bind stdout/stderr to other io.

### What version of nerdctl are you using?

main

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

**Comment by AkihiroSuda** [comment]
> A side note: not sure why this is an error in nerdctl/containerd if the folder doesn't exist. docker seems just creates the folder if not exist.


This is a bug 😓

**Comment by djdongjin** [comment]
close as the issue is resolved by #1943, and the other bind volume issue is tracked in #1945
